nix/nix.go

```diff
@@ -176,6 +176,24 @@ func (ctx *NixContext) GetBuildShell(deploymentPath string) (buildShell *string,
 	return buildShell, nil
 }
 
+func (ctx *NixContext) EvalHosts(deploymentPath string, attribute string) {
+	args := []string{ctx.EvalMachines,
+		"--arg", "networkExpr", deploymentPath,
+		"--eval", "--strict", "-A", attribute}
+
+	cmd := exec.Command("nix-instantiate", args...)
+	utils.AddFinalizer(func() {
+		if (cmd.ProcessState == nil || !cmd.ProcessState.Exited()) && cmd.Process != nil {
+			_ = cmd.Process.Signal(syscall.SIGTERM)
+		}
+	})
+
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Run()
+	// TODO: add error handling + finalizer(?)
```

Comment:
Yes, please react to the process erroring out so that an unsuccesful `nix-instantiate` does not result in a return code of 0. You're already passing stderr directly, which I think is fine, so any error message will be present already.